### ISSUE TITLE
apk with sqlite3 python3 kivy No module named '_sqlite3'

### ISSUE THREAD

**Issue by allaboutmikey** [kivy/python-for-android#1564]
Ran into problems with making an APK, mostly due to incorrect software and api versions. The last of those (unable to find a valid dependency graph) was solved over in https://github.com/kivy/python-for-android/issues/1222

Now my APK compiles fine: and crashes on load. The problem is it can't find the _sqlite3 module. The latest pythonforandroid has a recipe for sqlite3 which seems to compile just fine. It produces a libsqlite3.so file which is included in my APK (checked), but this doesn't seem to be what python is looking for when the app loads on the device.
I am able to run the rest of the app if I remove the import and use of the sqlite3 module, but it is critical to most of the functionality in the final version that sqlite3 work.

I have tried removing from blacklist.txt (wherever I could find one, there were several) any mention of sqlite3 files: no change.
I have tried whitelisting sqlite3 files in buildozer.spec: no change.
I have tried completely removing the .buildozer directory and allowing it all to re-downlooad/install/compile: no change.
Love some help if anyone has ideas.

Edit:
Just found in MyProject/.buildozer/android/platform/build/build/other_builds/python3/armeabi-v71__ndk_target_21/python3/android_build/config.log a mention of sqlite3 that looks like:
```
configure:10267: checking for --enable-loadable-sqlite-extensions
configure:10277: result: no
```
Does this mean sqlite is not being built into python3? Not sure how I would change this as buildozer is doing all the p4a stuff for me.

Edit 2:
Further research tells me that option is not required to get sqlite3 working.

**Comment by opacam** [comment]
The sqlite3 and openssl libs still are not python3's compatible in the current master branch. If you want to use them, you may wanna try the branch of the pr #1537, which enables python3 support for the mentioned libs.

**Comment by JonasT** [comment]
Note: to use sqlite3, put it into `--requirements` and it needs to go *before* "python3" in your requirements list (otherwise the core module for it won't be built)

**Comment by homdx** [comment]
Thank you, @JonasT for reccomendation.
> Note: to use sqlite put "sqlite3" into `--requirements` and it needs to go _before_ "python3" in your requirements list (otherwise the core module for it won't be built)

I checked today, and build apk with python3 with py3-api28ndk21 you tools with sqlite3 module https://github.com/JonasT/p4a-build-spaces
and build with p4a

**Comment by ApplySci** [comment]
> Note: to use sqlite put "sqlite3" into `--requirements` and it needs to go _before_ "python3" in your requirements list (otherwise the core module for it won't be built)

Oh, I didn't know that the order mattered at all. Is there some documentation on required order? Should python3 always come last, or does it need to be before kivy, or what?

**Comment by hackalog** [comment]
That's odd. Doesn't p4a build a dependency graph out of that information, reordering things anyway?

**Comment by JonasT** [comment]
@hackalog I don't think so, because currently sqlite3 is not a dependency. IMHO it should be (see #1576 ) but right now some of the non-vital core modules are only built when they happen to be around when the recipe is processed, and otherwise simply aren't

**Comment by Ham-Merhead** [comment]
I tried to build all the user modules for python3, kivy. Of course a successful
apk build does not necessarily imply the apk contents are correct!

The following dependencies are required in addition to those for a simple .py build
(for which I use https://groups.google.com/forum/#!topic/kivy-users/Lf4zlYmLVPo  )
`sudo apt-get install -y autoconf libtool cmake libffi-dev pkg-config
`

I use python-for-android (master, installed 1/12/2019) 
I tested each with a placeholder main.py and:
`p4a apk --requirements=<module>,python3,kivy --arch=armeabi-v7a --ndk_version 17c --android_api 28 --ndk-api 21`

The following modules successfully build an apk:

```
['Pillow', 'audiostream', 'cffi', 'cryptography', 'enum34', 'ffmpeg',
'ffpyplayer_codecs', 'freetype', 'harfbuzz', 'idna', 'ipaddress', 'jedi',
'jpeg', 'leveldb', 'libcurl', 'libexpat', 'libffi', 'libiconv', 'libogg',
'libsecp256k1', 'libshine', 'libvorbis', 'libx264', 'libxml2', 'libxslt',
'lxml','numpy', 'openal','openssl', 'pbkdf2', 'png', 'pycparser',
'pycryptodome', 'pyjnius', 'pysdl2','requests', 'sdl2_image', 'sdl2_mixer',
'sdl2_ttf', 'six', 'snappy', 'spine','sqlite3']
```

BUT, of these, the following fail with error ".. _ctypes..""
if module order is changed  --requirements=python3,kivy,\<module\>

```
['Pillow','cffi','cryptography','enum34','idna','ipaddress','pbkdf2',
'pycparser','pycryptodome','requests','spine']

```
Which confirms that order can be important.

The following do not build because they have no dependency on 'python3', but
do have a dependency on some combination of 'python2'/'hostpython2'/
'python3crystax'/'hostpython3'

```
['apsw','atom','babel','boost','cdecimal','cherrypy','coverage','cymunk',
'dateutil','decorator','doubleratchet','enaml','ethash','evdev','feedparser',
'gevent','gevent-websocket','greenlet','groestlcoin_hash','icu','ifaddrs',
'kiwisolver','libglob','libnacl','libpq','libsodium','libtorrent',
'libtribler','libzbar','libzmq','m2crypto','msgpack-python','netifaces',
'omemo','omemo-backend-signal','pil','preppy','psycopg2','pyaml','pyasnl',
'pycrypto','pyethereum','pyicu','pyleveldb','pymunk','pynacl','pyogg',
'pyopenal','pyopenssl','pyproj','pyrxp','pysha3','pyusb','pyyaml','pyzmq',
'regex','ruamel.yaml','scrypt','secp256k1','shapely','simple-crypt',
'sqlalchemy','storm','sympy','ujson','vispy','websocket-client','wsaccl',
'x3dh','xeddsa','zbar','zeroconf','zope','zope_interface']
```

The following does not build because of a dependency I can't identify:
`['flask']`

The following do not build because they have a dependency on 'kivy' !
`['kivent_core','kivent_cymunk','kivent_particles','kivent_polygen']`

The following does not build because:
'p4a.cmake': No such file or directory
`['libmysqlclient']`

The following does not build because of cmake errors
`['opencv']`

The following does not build because (seems strange):
crypt.h: No such file or directory
`['reportlab']`

The following does not build because has dependency on 'incremental', 'constantly'
`['twisted']`

The following does not build because NDK v18 needed
`['vlc']   `

**Comment by inclement** [comment]
Closing as hopefully fixed by #1617, please feel free to reopen if it does still happen.
